test/Microsoft.ML.Tests/Transformers/ValueMappingTests.cs

```diff
@@ -320,6 +320,32 @@ public void ValueMappingValuesAsStringKeyTypes()
             Assert.Equal<uint>(0, fValue);
         }
 
+        [Fact]
+        public void ValueMappingValuesAsKeyTypesReverseLookup()
+        {
+            var data = new[] { new TestClass() { A = "bar", B = "test", C = "notfound" } };
+            var dataView = ComponentCreation.CreateDataView(Env, data);
+
+            var keys = new List<ReadOnlyMemory<char>>() { "foo".AsMemory(), "bar".AsMemory(), "test".AsMemory(), "wahoo".AsMemory() };
+
+            // Generating the list of strings for the key type values, note that foo1 is duplicated as intended to test that the same index value is returned
+            var values = new List<ReadOnlyMemory<char>>() { "foo1".AsMemory(), "foo2".AsMemory(), "foo1".AsMemory(), "foo3".AsMemory() };
+
+            var estimator = new ValueMappingEstimator<ReadOnlyMemory<char>, ReadOnlyMemory<char>>(Env, keys, values, true, new[] { ("A", "D"), ("B", "E"), ("C", "F") })
+                            .Append(new KeyToValueMappingEstimator(Env, ("D","DOutput")));
```

Comment:
```suggestion
                            .Append(new KeyToValueMappingEstimator(Env, ("D", "DOutput")));
```
Please format all files touched.

Reply:
Thanks, they are both formatted now.

---
In reply to: [247284278](https://github.com/dotnet/machinelearning/pull/2098#discussion_r247284278) [](ancestors = 247284278)